helper/netsim-throughput-charting.h

```diff
@@ -0,0 +1,220 @@
+/* -*-  Mode: C++; c-file-style: "gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * NIST-developed software is provided by NIST as a public
+ * service. You may use, copy and distribute copies of the software in
+ * any medium, provided that you keep intact this entire notice. You
+ * may improve, modify and create derivative works of the software or
+ * any portion of the software, and you may copy and distribute such
+ * modifications or works. Modified works should carry a notice
+ * stating that you changed the software and should note the date and
+ * nature of any such change. Please explicitly acknowledge the
+ * National Institute of Standards and Technology as the source of the
+ * software.
+ *
+ * NIST-developed software is expressly provided "AS IS." NIST MAKES
+ * NO WARRANTY OF ANY KIND, EXPRESS, IMPLIED, IN FACT OR ARISING BY
+ * OPERATION OF LAW, INCLUDING, WITHOUT LIMITATION, THE IMPLIED
+ * WARRANTY OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE,
+ * NON-INFRINGEMENT AND DATA ACCURACY. NIST NEITHER REPRESENTS NOR
+ * WARRANTS THAT THE OPERATION OF THE SOFTWARE WILL BE UNINTERRUPTED
+ * OR ERROR-FREE, OR THAT ANY DEFECTS WILL BE CORRECTED. NIST DOES NOT
+ * WARRANT OR MAKE ANY REPRESENTATIONS REGARDING THE USE OF THE
+ * SOFTWARE OR THE RESULTS THEREOF, INCLUDING BUT NOT LIMITED TO THE
+ * CORRECTNESS, ACCURACY, RELIABILITY, OR USEFULNESS OF THE SOFTWARE.
+ *
+ * You are solely responsible for determining the appropriateness of
+ * using and distributing the software and you assume all risks
+ * associated with its use, including but not limited to the risks and
+ * costs of program errors, compliance with applicable laws, damage to
+ * or loss of data, programs or equipment, and the unavailability or
+ * interruption of operation. This software is not intended to be used
+ * in any situation where a failure could cause risk of injury or
+ * damage to property. The software developed by NIST employees is not
+ * subject to copyright protection within the United States.
+ *
+ * Author: Nihar Kapasi <niharkkapasi@gmail.com>
+ */
+
+#ifndef THROUGHPUT_CHARTING_H
+#define THROUGHPUT_CHARTING_H
+
+#include <ns3/application-container.h>
+#include <ns3/nstime.h>
+#include <ns3/packet.h>
+#include <ns3/ptr.h>
+#include <ns3/type-id.h>
+
+#include "ns3/orchestrator.h"
+#include "ns3/throughput-sink.h"
+
+
+namespace ns3::netsimulyzer
+{
+
+using namespace std;
+
+/*
+ * traceSourceType refers to a string representation of the trace sink type
+ * The last param to AddTraceSource conventionally - although can use a user defined string here
+ * 
+ * traceSourceName refers to the name that corresponds to the Tx/Rx trace source propery name used in a TraceConnect
+ */
+struct TraceSourceInfo
+{
+  std::string traceSourceType;
+  std::string traceSourceName;
+};
+
+/*
+ * Class that creates throughput graphs for each of the supplied apps 
+ * ASSUMPTION: ip address is of type ipv4 (takes the first ip address in list for node)
```

Comment:
Evan, I assume we may want to remove this IPv4 restriction?  We could merge this and open up an issue to extend it later.